proxy/conn_test.go

```diff
@@ -0,0 +1,189 @@
+package proxy
+
+import (
+	"bufio"
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/websocket"
+	"github.com/matrix-org/matrix-websockets-proxy/mocks/io"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+// testSendMessage tests that Conn.SendMessage causes a text message to be
+// written to the web socket
+func TestSendMessage(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	wg := &sync.WaitGroup{}
+
+	// make the websocket
+	recvPipeRdr, _ := io.Pipe()
+	writer, ws, err := makeWsConn(recvPipeRdr, ctrl)
+	if err != nil {
+		t.Errorf("Expected no error, got '%v'", err)
```

Comment:
t.Fatalf exists to stop the test and report an error; no need to t.Errrof ; return